test/ivan-vole/json-transformer.spec.ts

```diff
@@ -0,0 +1,144 @@
+}
+
+const transformJsonWithCustomResolvers = (result: IOutputResult) => transformIOuputToJson(result, customResolveIDomTextNode, customResolveIDomHtmlNode)
+
+describe("Json Transfomer Tests", () => {
+    it("Test empty", () => {
+        const testValue: IOutputResult = {
+            children: []
+        }
+
+        const output = transformJsonWithCustomResolvers(testValue);
+
+        expect(output).toEqual([]);
+    })
+
+    it("Test only IDomTextNode", () => {
+        const testValue: IOutputResult = {
+            children: [
+                {
+                    type:"text",
+                    content:"test value",
+                }
+            ]
+        }
+
+        const output = transformJsonWithCustomResolvers(testValue)[0];
+
+        const expectedOutput = {
+            text: "test value"
+        };
+
+
+        expect(output).toEqual(expectedOutput);
+    })
+
+    it("Test only IDomHtmlNode", () => {
+        const testValue: IOutputResult = {
+            children: [
+                {
+                    type:"tag",
+                    tagName:"p",
+                    attributes: {},
+                    children: []
+                }
+            ]
+        }
+
+        const output = transformJsonWithCustomResolvers(testValue)[0];
+
+        const expectedOutput = {
+            tag: "p",
+            children: []
+        };
+
+
+        expect(output).toEqual(expectedOutput);
+    })
+
+    it("Test RichText", () => {
+        const parsed = (new RichTextBrowserParser()).parse(dummy.value);
+        const trasnformed = transformJsonWithCustomResolvers(parsed);
+        
+        const expectedOutput= JSON.parse(`
+        [
+            {
+               "tag":"p",
+               "children":[
+                  {
+                     "text":"Test from rich text"
+                  }
+               ]
+            },
+            {
+               "tag":"figure",
+               "imageId":"7d866175-d3db-4a02-b0eb-891fb06b6ab0",
+               "children":[
+                  {
+                     "tag":"img",
+                     "src":"https://assets-eu-01.kc-usercontent.com:443/6d864951-9d19-0138-e14d-98ba886a4410/236ecb7f-41e3-40c7-b0db-ea9c2c44003b/sharad-bhat-62p19OGT2qg-unsplash.jpg",
+                     "alt":"",
+                     "children":[
+                        
+                     ]
+                  }
+               ]
+            },
+            {
+               "tag":"object",
+               "codeName":"e53bff4f_0f6e_0168_a2fe_5ec0eaa032de",
+               "children":[]
+            }
+         ]`  
+        )
+
+        expect(trasnformed).toEqual(expectedOutput);
+    })
+})
```

Comment:
Try transform:
* html table to some json alternation
* HTML nested list to some json alternation 
* component & linked item